packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -64,17 +102,23 @@ export const TagGroup = ((props: SpectrumTagGroupProps) => {
           styleProps.className
         )
       }
-      {...tagGroupProps}>
-      <TagGroupContext.Provider
-        value={{
-          isRemovable: isReadOnly ? false : isReadOnly,
-          isDisabled,
-          onRemove: isReadOnly ? null : removeAll,
-          validationState,
-          role: 'gridcell'
-        }}>
-        {children}
-      </TagGroupContext.Provider>
+      tabIndex={isDisabled ? -1 : 0}
```

Comment:
```suggestion
      tabIndex={isDisabled ? -1 : gridProps.tabIndex}
```
Perhaps even move the `useGrid` call into `useTagGroup` and handle the tab index resolution in `useTagGroup` instead